src/number_format.js

```diff
@@ -84,6 +84,7 @@ const defaultProps = {
   suffix: '',
   allowNegative: true,
   type: 'text',
+  value: '',
```

Comment:
Test failing because of this as if value is not provided we treat it as an uncontrolled component.